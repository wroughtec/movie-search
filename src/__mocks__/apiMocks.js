export const searchMock = {
  page: 1,
  total_results: 140,
  total_pages: 7,
  results: [
    {
      vote_count: 9703,
      id: 11,
      video: false,
      vote_average: 8.2,
      title: 'Star Wars',
      popularity: 44.515,
      poster_path: '/btTdmkgIvOi0FFip1sPuZI2oQG6.jpg',
      original_language: 'en',
      original_title: 'Star Wars',
      genre_ids: [],
      backdrop_path: '/4iJfYYoQzZcONB9hNzg0J0wWyPH.jpg',
      adult: false,
      overview:
        'Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire. Venturesome Luke Skywalker and dashing captain Han Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and restore peace and justice in the Empire.',
      release_date: '1977-05-25'
    }
  ]
};

export const configMock = {
  images: {
    base_url: 'http://image.tmdb.org/t/p/',
    secure_base_url: 'https://image.tmdb.org/t/p/',
    backdrop_sizes: ['w300', 'w780', 'w1280', 'original'],
    logo_sizes: ['w45', 'w92', 'w154', 'w185', 'w300', 'w500', 'original'],
    poster_sizes: ['w92', 'w154', 'w185', 'w342', 'w500', 'w780', 'original'],
    profile_sizes: ['w45', 'w185', 'h632', 'original'],
    still_sizes: ['w92', 'w185', 'w300', 'original']
  },
  change_keys: []
};

export const genresMock = [{ id: 1, name: 'new' }];
