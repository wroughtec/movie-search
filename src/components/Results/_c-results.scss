@import '../../scss/_mixins/m-container';

.c-results {
  @include container;
}

.c-results__container {
  display: grid;

  grid-gap: 2rem;
  grid-template-columns: repeat(auto-fit, 18.75rem);

  justify-content: center;
}
