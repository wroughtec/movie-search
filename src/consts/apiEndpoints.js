// @flow

export const searchEndpoint = 'search/movie';
export const configEndpoint = 'configuration';
export const genresEndpoint = 'genre/movie/list';
