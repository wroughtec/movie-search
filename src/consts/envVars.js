// @flow

export const BASE_URL: ?string = process.env.API_URL;
export const KEY: ?string = process.env.API_KEY;
