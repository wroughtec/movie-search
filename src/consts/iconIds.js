// @flow

export const spinnerIcon = 'spinner';
