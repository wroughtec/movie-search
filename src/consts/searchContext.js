// @flow

export const searchContext: SearchContextType = {
  loading: true,
  imageBaseUrl: '',
  sm: '',
  md: '',
  lg: '',
  xl: '',
  genres: [],
  searchTerms: '',
  searchResults: {},
  searchPageNo: 1,
  handleMovieSubmit() {},
  handleSearchChange() {},
  popularResults: {},
  popularPageNo: 1,
  updateSearchTerms() {}
};
