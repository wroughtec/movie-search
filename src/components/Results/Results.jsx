// @flow

import React, { Component } from 'react';
import { Spinner } from 'components/Spinner/Spinner';
import { Card } from 'components/Card/Card';
import './_c-results.scss';

type Props = {
  searchParams: SearchContextType,
  searchTerm?: ?string
};
export class Results extends Component<Props> {
  static defaultProps = {
    searchTerm: ''
  };

  async componentDidMount() {
    const {
      searchTerm,
      searchParams: { updateSearchTerms, loadMovies }
    } = this.props;

    if (searchTerm) {
      await updateSearchTerms(searchTerm);
      loadMovies();
    }
  }

  displayCards = () => {
    const { searchParams, popular } = this.props,
      { searchResults, imageBaseUrl, popularResults, lg } = searchParams;

    let cards = <div className="c-results__no-results">No results</div>,
      movieResults = searchResults;

    if (popular) {
      movieResults = popularResults;
    }
    if (movieResults && movieResults.results && movieResults.results.length) {
      cards = movieResults.results.map(card => {
        const { title, id, release_date: releaseDate, poster_path: posterPath, overview, genre_ids: genreIds } = card;
        return (
          <Card
            key={id}
            title={title}
            releaseDate={releaseDate}
            overview={overview}
            path={posterPath}
            imageBaseUrl={imageBaseUrl}
            size={lg}
            genreIds={genreIds}
            id={id}
          />
        );
      });
    }

    return cards;
  };

  render() {
    const { searchParams } = this.props,
      { loading } = searchParams,
      cards = this.displayCards();

    return (
      <section className="c-results">
        {loading && <Spinner />}
        {!loading && <div className="c-results__container">{cards}</div>}
      </section>
    );
  }
}
