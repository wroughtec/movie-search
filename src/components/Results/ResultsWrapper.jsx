// @flow

import React from 'react';
import { Consumer } from 'components/SearchContext/SearchContext';
import { Results } from './Results';

export const ResultsWrapper = (props: any) => (
  <Consumer>{(context: SearchContextType) => <Results {...props} searchParams={context} />}</Consumer>
);
