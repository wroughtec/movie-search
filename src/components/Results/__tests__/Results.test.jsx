import React from 'react';
import { shallow } from 'enzyme';
import { Results } from 'components/Results/Results';
import { searchContext } from 'consts/searchContext';
import { searchMock } from 'mocks/apiMocks';

describe('Results', () => {
  const defaultProps = {
      searchParams: {
        ...searchContext
      }
    },
    generateProps = ({ newProps, searchProps }) => ({
      searchParams: {
        ...searchContext,
        ...searchProps
      },
      ...newProps
    });
  test('Renders', () => {
    const component = shallow(<Results {...defaultProps} />);

    expect(component).toMatchSnapshot();
  });

  test('include search term', () => {
    const props = generateProps({
      newProps: { searchTerm: 'star wars' },
      searchProps: { loading: false, updateSearchTerms: jest.fn(), loadMovies: jest.fn() }
    });
    const component = shallow(<Results {...props} />);
    component.instance().componentDidMount();

    expect(component).toMatchSnapshot();
  });

  test('update searchTerms when searchTerm supplied', async () => {
    const updateSearchTerms = jest.fn(),
      loadMovies = jest.fn(),
      props = generateProps({
        newProps: { searchTerm: 'star wars' },
        searchProps: { updateSearchTerms, loadMovies }
      });

    const component = shallow(<Results {...props} />);

    await component.instance().componentDidMount();

    expect(updateSearchTerms).toHaveBeenCalled();
    expect(loadMovies).toHaveBeenCalled();
  });

  test('shows cards container', () => {
    const props = generateProps({ searchProps: { loading: false } });
    const component = shallow(<Results {...props} />);

    expect(component).toMatchSnapshot();
    expect(component.find('.c-results__container')).toHaveLength(1);
  });

  test('shows cards', () => {
    const props = generateProps({ searchProps: { loading: false, searchResults: searchMock } });
    const component = shallow(<Results {...props} />);

    expect(component).toMatchSnapshot();
  });
});
