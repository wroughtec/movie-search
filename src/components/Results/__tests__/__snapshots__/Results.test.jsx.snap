// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Results
    searchParams={
      Object {
        "genres": Array [],
        "handleMovieSubmit": [Function],
        "handleSearchChange": [Function],
        "imageBaseUrl": "",
        "lg": "",
        "loading": true,
        "md": "",
        "popularPageNo": 1,
        "popularResults": Object {},
        "searchPageNo": 1,
        "searchResults": Object {},
        "searchTerms": "",
        "sm": "",
        "updateSearchTerms": [Function],
        "xl": "",
      }
    }
    searchTerm=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Spinner />,
        false,
      ],
      "className": "c-results",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Spinner />,
          false,
        ],
        "className": "c-results",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Results include search term 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Results
    searchParams={
      Object {
        "genres": Array [],
        "handleMovieSubmit": [Function],
        "handleSearchChange": [Function],
        "imageBaseUrl": "",
        "lg": "",
        "loadMovies": [MockFunction],
        "loading": false,
        "md": "",
        "popularPageNo": 1,
        "popularResults": Object {},
        "searchPageNo": 1,
        "searchResults": Object {},
        "searchTerms": "",
        "sm": "",
        "updateSearchTerms": [MockFunction] {
          "calls": Array [
            Array [
              "star wars",
            ],
            Array [
              "star wars",
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "xl": "",
      }
    }
    searchTerm="star wars"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <div
          className="c-results__container"
        >
          <div
            className="c-results__no-results"
          >
            No results
          </div>
        </div>,
      ],
      "className": "c-results",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="c-results__no-results"
          >
            No results
          </div>,
          "className": "c-results__container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No results",
            "className": "c-results__no-results",
          },
          "ref": null,
          "rendered": "No results",
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <div
            className="c-results__container"
          >
            <div
              className="c-results__no-results"
            >
              No results
            </div>
          </div>,
        ],
        "className": "c-results",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="c-results__no-results"
            >
              No results
            </div>,
            "className": "c-results__container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No results",
              "className": "c-results__no-results",
            },
            "ref": null,
            "rendered": "No results",
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Results shows cards 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Results
    searchParams={
      Object {
        "genres": Array [],
        "handleMovieSubmit": [Function],
        "handleSearchChange": [Function],
        "imageBaseUrl": "",
        "lg": "",
        "loading": false,
        "md": "",
        "popularPageNo": 1,
        "popularResults": Object {},
        "searchPageNo": 1,
        "searchResults": Object {
          "page": 1,
          "results": Array [
            Object {
              "adult": false,
              "backdrop_path": "/4iJfYYoQzZcONB9hNzg0J0wWyPH.jpg",
              "genre_ids": Array [],
              "id": 11,
              "original_language": "en",
              "original_title": "Star Wars",
              "overview": "Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire. Venturesome Luke Skywalker and dashing captain Han Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and restore peace and justice in the Empire.",
              "popularity": 44.515,
              "poster_path": "/btTdmkgIvOi0FFip1sPuZI2oQG6.jpg",
              "release_date": "1977-05-25",
              "title": "Star Wars",
              "video": false,
              "vote_average": 8.2,
              "vote_count": 9703,
            },
          ],
          "total_pages": 7,
          "total_results": 140,
        },
        "searchTerms": "",
        "sm": "",
        "updateSearchTerms": [Function],
        "xl": "",
      }
    }
    searchTerm=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <div
          className="c-results__container"
        >
          <Card
            genreIds={Array []}
            imageBaseUrl=""
            overview="Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire. Venturesome Luke Skywalker and dashing captain Han Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and restore peace and justice in the Empire."
            path="/btTdmkgIvOi0FFip1sPuZI2oQG6.jpg"
            releaseDate="1977-05-25"
            size=""
            title="Star Wars"
          />
        </div>,
      ],
      "className": "c-results",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              genreIds={Array []}
              imageBaseUrl=""
              overview="Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire. Venturesome Luke Skywalker and dashing captain Han Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and restore peace and justice in the Empire."
              path="/btTdmkgIvOi0FFip1sPuZI2oQG6.jpg"
              releaseDate="1977-05-25"
              size=""
              title="Star Wars"
            />,
          ],
          "className": "c-results__container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "11",
            "nodeType": "function",
            "props": Object {
              "genreIds": Array [],
              "imageBaseUrl": "",
              "overview": "Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire. Venturesome Luke Skywalker and dashing captain Han Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and restore peace and justice in the Empire.",
              "path": "/btTdmkgIvOi0FFip1sPuZI2oQG6.jpg",
              "releaseDate": "1977-05-25",
              "size": "",
              "title": "Star Wars",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <div
            className="c-results__container"
          >
            <Card
              genreIds={Array []}
              imageBaseUrl=""
              overview="Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire. Venturesome Luke Skywalker and dashing captain Han Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and restore peace and justice in the Empire."
              path="/btTdmkgIvOi0FFip1sPuZI2oQG6.jpg"
              releaseDate="1977-05-25"
              size=""
              title="Star Wars"
            />
          </div>,
        ],
        "className": "c-results",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Card
                genreIds={Array []}
                imageBaseUrl=""
                overview="Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire. Venturesome Luke Skywalker and dashing captain Han Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and restore peace and justice in the Empire."
                path="/btTdmkgIvOi0FFip1sPuZI2oQG6.jpg"
                releaseDate="1977-05-25"
                size=""
                title="Star Wars"
              />,
            ],
            "className": "c-results__container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "11",
              "nodeType": "function",
              "props": Object {
                "genreIds": Array [],
                "imageBaseUrl": "",
                "overview": "Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire. Venturesome Luke Skywalker and dashing captain Han Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and restore peace and justice in the Empire.",
                "path": "/btTdmkgIvOi0FFip1sPuZI2oQG6.jpg",
                "releaseDate": "1977-05-25",
                "size": "",
                "title": "Star Wars",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Results shows cards container 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Results
    searchParams={
      Object {
        "genres": Array [],
        "handleMovieSubmit": [Function],
        "handleSearchChange": [Function],
        "imageBaseUrl": "",
        "lg": "",
        "loading": false,
        "md": "",
        "popularPageNo": 1,
        "popularResults": Object {},
        "searchPageNo": 1,
        "searchResults": Object {},
        "searchTerms": "",
        "sm": "",
        "updateSearchTerms": [Function],
        "xl": "",
      }
    }
    searchTerm=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <div
          className="c-results__container"
        >
          <div
            className="c-results__no-results"
          >
            No results
          </div>
        </div>,
      ],
      "className": "c-results",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="c-results__no-results"
          >
            No results
          </div>,
          "className": "c-results__container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No results",
            "className": "c-results__no-results",
          },
          "ref": null,
          "rendered": "No results",
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <div
            className="c-results__container"
          >
            <div
              className="c-results__no-results"
            >
              No results
            </div>
          </div>,
        ],
        "className": "c-results",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="c-results__no-results"
            >
              No results
            </div>,
            "className": "c-results__container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No results",
              "className": "c-results__no-results",
            },
            "ref": null,
            "rendered": "No results",
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
